services/src/main/java/org/keycloak/validation/DefaultClientValidationProvider.java

```diff
@@ -180,6 +199,24 @@ private void checkUri(FieldMessages field, String url, ValidationContext<ClientM
         }
     }
 
+    private void checkUriLogo(FieldMessages field, String url, ValidationContext<ClientModel> context) {
+        if (url == null || url.isEmpty()) {
+            return;
+        }
+
+        try {
+            URI uri = new URI(url);
+
+            if (uri.getScheme() != null &&  uri.getScheme().equals("javascript")) {
```

Comment:
This should be more strict: "Schemes other than “https”, “http”, or “data” SHOULD NOT be used."
according to https://docs.oasis-open.org/security/saml/Post2.0/sstc-saml-metadata-ui/v1.0/os/sstc-saml-metadata-ui-v1.0-os.html#__RefHeading__10407_1021935550

Reply:
It is not mandatory. Moreover, as I know OIDC specification has nothing similar. I want validator to be common for OIDC and SAML ( as in other fields) and to be similar to other fields validator - only logo_uri can have data scheme.
However, if Keycloak team want, I can add proposed validator only for SAML client.